### ISSUE TITLE
[FR] Allow theming a step when invalid+dirty

### ISSUE THREAD

**Issue by Goodwine** [angular/components#8950]
#### Proposal:
Allow theming a step when invalid+dirty. Perhaps it could default to the current behavior and make it opt in, similar to how buttons are colorless unless we set color="primary".

#### What is the expected behavior?
```html
<mat-step errorColor="warn" errorIcon="warning" [stepControl]="formGroup"></mat-step>
```
This code should automatically set the icon to a warning instead of a circle with a number/checkmark/pencil inside it. And.. it should color both text and icon (or maybe only the icon) to the theme's warn color.

#### What is the current behavior?
There is no visual indicator when a step is invalid. Except if we use a fancier step label, but the icon remains untouched.

#### What are the steps to reproduce?
https://angular-5zj5lo-jqlqug.stackblitz.io
1- Make form linear
2- Type something on the name field.
3- Erase it.

#### What is the use-case or motivation for changing an existing behavior?
Imagine 3 steps: A, B, C. Step B calls a service that could update the input data for A. The new data now marks step A as being invalid and no matter how many times the user clicks the submit button, the form won't be submitted because step A is failing but the user won't be able to tell.

One argument against this is "well. you should refactor your stepper", but this is not an option at the moment, Step A doesn't depend on Step B, however Step B refreshes the form's input data which could cause Step A to be invalid.

#### Is there anything else we should know?
A better interface for such feature would be great, I don't like mine.

**Comment by Goodwine** [comment]
@mmalerba

**Comment by Goodwine** [comment]
Just to add a bit more context, the [material stepper specs](https://material.io/guidelines/components/steppers.html#steppers-specs) do have this:

![image](https://user-images.githubusercontent.com/2022649/33907148-049daf8e-df39-11e7-86b0-e341d60a6147.png)

**Comment by mmalerba** [comment]
The current behavior definitely doesn't look right, we might not even need a new feature for this, maybe we can just use the error color when its in error state (and we already have a way to change the icons)

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
